db/migrate/20150716133712_add_unique_index_on_journals.rb

```diff
@@ -0,0 +1,89 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class AddUniqueIndexOnJournals < ActiveRecord::Migration
+  def up
+    cleanup_duplicate_journals
+    add_index :journals, [:journable_type, :journable_id, :version], unique: true
+  end
+
+  def down
+    remove_index :journals, [:journable_type, :journable_id, :version]
+  end
+
+  private
+
+  def cleanup_duplicate_journals
```

Comment:
Assignment Branch Condition size for cleanup_duplicate_journals is too high. [25.36/15]